test/chain-test.js

```diff
@@ -761,8 +766,44 @@ describe('Chain', function() {
       'mandatory-script-verify-flag-failed');
   });
 
-  it('should cleanup', async () => {
-    await miner.close();
-    await chain.close();
+  describe('Checkpoints', function() {
+    before(async () => {
+      const entry = await chain.getEntry(chain.tip.height - 5);
+      assert(Buffer.isBuffer(entry.hash));
+      assert(Number.isInteger(entry.height));
+
+      network.checkpointMap[entry.height] = entry.hash;
+      network.lastCheckpoint = entry.height;
+    });
+
+    after(async () => {
+      network.checkpointMap = {};
+      network.lastCheckpoint = 0;
+    });
+
+    it('will reject blocks before last checkpoint', async () => {
```

Comment:
This test is fragile to tree root updates breaking the call to `cpu.mineBlock` if the tree interval is crossed until https://github.com/handshake-org/hsd/issues/159 is fixed, although I don't expect that problem to ever be ran into. Tests in https://github.com/handshake-org/hsd/pull/305 are as well.